### PR TITLE
Step 1077 type param

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist

- [X] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires MINOR [version update](https://semver.org/)

### Context

Xcode 13 beta support, added missing platform type parameter.

Resolves: https://bitrise.atlassian.net/browse/STEP-1077

### Changes

- Added "Platform" (`platform`) input, defaults to `ios`.
- If using Xcode 13 or later, add the `--type` parameter to altool.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -31,6 +32,7 @@ type Config struct {
 
 	IpaPath           string `env:"ipa_path"`
 	PkgPath           string `env:"pkg_path"`
+	Platform          string `env:"platform,opt[ios,macos,tvos]"`
```
Comment (BirmacherAkos) [inline main.go:35]: Could we detect this instead?

Comment (BirmacherAkos) [reply main.go:35]: Idea: [**auto**,ios,macos,tvos]. With auto the step should try to figure out the platform.

File: main.go [hunk 2]
```diff
@@ -219,13 +273,23 @@ func main() {
...
 	additionalParams, err := shellquote.Split(cfg.AdditionalParams)
 	if err != nil {
 		failf("Failed to parse additional parameters, error: %s", err)
 	}
 
-	altoolParams := append([]string{"altool", "--upload-app", "-f", filePth}, authParams...)
+	uploadParams := []string{"--upload-app", "-f", filePth}
+	// Platform type parameter was introduced in Xcode 13
+	if xcodeVersion.MajorVersion >= 13 && !sliceutil.IsStringInSlice(typeKey, additionalParams) {
+		uploadParams = append(uploadParams, typeKey, getPlatformType(cfg.IpaPath, cfg.Platform))
```
Comment (lpusok) [inline main.go:288]:
```suggestion
		uploadParams = append(uploadParams, typeKey, getPlatformType(filePth, cfg.Platform))
```

Comment (mateherber) [reply main.go:288]: It is intended that I pass only the ipa path. If it is not ipa (pkg) I pass "" and it is handled in the function

File: main.go [hunk 3]
```diff
@@ -55,6 +59,49 @@ func (cfg Config) validateArtifact() error {
...
+		if err != nil {
+			return fallback()
+		}
+		platform, ok := plist.GetString("DTPlatformName")
+		if !ok {
+			return fallback()
+		}
+		switch platform {
+		case "appletvos", "appletvsimulator":
+			return "appletvos"
```
Comment (lpusok) [inline main.go:89]: NOTE: macos, appletvos and ios could be extracted to a constant (or enum).

**Comment by lpusok** [comment]
Looks ok, CI needs to be fixed/migrated tough.

**Pull request merged**